src/Model/Resolver/Page.php

```diff
@@ -69,35 +68,29 @@ public function resolve(
         array $value = null,
         array $args = null
     ) : Value {
-
         $result = function () use ($args) {
             $pageId = $this->getPageId($args);
             $pageData = $this->getPageData($pageId);
-
             return $pageData;
         };
         return $this->valueFactory->create($result);
     }
-
     /**
      * @param array $args
      * @return int
      * @throws GraphQlInputException
      */
     private function getPageId(array $args): int
     {
+        $storeId = $this->storeManager->getStore()->getId();
         if (isset($args['id'])) {
             return (int)$args['id'];
         }
-
         if (isset($args['url_key'])) {
-            return (int)$this->pageFactory->create()->load($args['url_key'])->getId();
+            return (int)$this->pageFactory->create()->setStoreId($storeId)->load($args['url_key'])->setStoreId($storeId)->getId();
```

Comment:
Why store I'd is set twice? Also, please move each -> to new pinez so it is easier to read :)

Reply:
In the past I had a problem with single set store id on factory, twice set is safety :D